strata.default.lua

```diff
@@ -31,78 +31,70 @@ strata.set_bindings {
-	-- what's above.
+-- strata.set_rules {
+-- 	-- Always bind firefox to the workspace 1, before the window is opened
+-- 	{
+-- 		triggers = { { event = "win_open_pre", class_name = "firefox" } },
+-- 		action = function(window) window.send_to_workspace(1) end,
+-- 	},
+--
+-- 	-- A single action that can be ran from multiple triggers.
+-- 	{
+-- 		triggers = {
+-- 			-- Set mpv to floating, always
+-- 			{ event = "win_open_pre", class_name = "mpv" },
+-- 			-- Set terminals to floating only if on the workspace 1 (where we put firefox previously)
+-- 			{ event = "win_open_pre", workspace = 1, class_name = "kitty" },
+-- 		},
+-- 		action = function(window) window.set_floating() end,
+-- 	},
 
-	-- This one does the same thing as the first rule
-	strata.rules.bind_to_workspace(1, "firefox"),
-
-	-- Nothing preventing us from some more sugar, to make things even less verbose:
-	strata.rules.bind_to_workspace {
-		{ 1, "firefox" },
-		{ 2, "neovide" },
-		{ 10, "slack" },
-	},
+-- Here comes the nice stuff: helper functions! This is important, as these functions help make the config file much
+-- easier to write for the majority of users. Ideally, most people won't ever need to write something detailed like
+-- what's above.
 
-	-- This does the same thing as the first trigger of the second rule
-	strata.rules.set_floating("mpv"),
-}
+-- This one does the same thing as the first rule
+-- strata.rules.bind_to_workspace(1, "firefox"),
+--
+-- -- Nothing preventing us from some more sugar, to make things even less verbose:
+-- strata.rules.bind_to_workspace({
+-- 	{ 1, "firefox" },
+-- 	{ 2, "neovide" },
+-- 	{ 10, "slack" },
+-- }),
+--
+-- -- This does the same thing as the first trigger of the second rule
+-- strata.rules.set_floating("mpv"),
+-- }
 
 -- Same as above, this is a function call. I think as it stands it should *overwrite* the whole config every time it's
 -- called. For a "merge" behavior, we could have a `strata.update_config()` function instead. This can one day be useful
 -- for plugins.
 strata.set_config {
 	autostart = {
-		{ "kitty", "--title", "Terminal" },
+		{ "kitty --title Terminal" },
 		{ "kagi" },
 	},
 	general = {
-		workspaces = 1,
+		workspaces = 10,
 		gaps_in = 8,
 		gaps_out = 12,
 		kb_repeat = { 500, 250 },
 	},
 	decorations = {
-		border = {
-			width = 2,
-			active = "#FFF",
-			inactive = "#131418",
-			radius = 5,
-		},
-		window = {
-			opacity = 0.9,
-		},
-		blur = {
-			enabled = true,
-			size = 2,
-			passes = 3,
-			optimize = true,
-		},
-		shadow = {
-			enabled = true,
-			size = 2,
-			blur = 3,
-			color = "#FFF",
-		},
+		border_width = 2,
+		border_active = "#FFF",
+		border_inactive = "#131418",
+		border_radius = 5,
+		window_opacity = 0.9,
+		blur_enable = true,
+		blur_size = 2,
+		blur_passes = 3,
+		blur_optimize = true,
+		shadow_enabled = true,
+		shadow_size = 2,
+		shadow_blur = 3,
+		shadow_color = "#FFF",
```

Comment:
It's a matter of taste, but I'm not sure I like this. It's more verbose, and it'll look a bit messier in rust as well, while we could have dedicated structs for every subcategory otherwise (and parsing handled by serde).

No reason to block this because I don't like it though, but that's how I see it.

Reply:
Its what I thought too. I'll work on a different config that I have in mind (its almost the same, maybe its the same lol) and open a PR